lib/msal-core/test/utils/RequestUtils.spec.ts

```diff
@@ -4,63 +4,131 @@ import { CryptoUtils } from "../../src/utils/CryptoUtils";
 import { AuthenticationParameters } from "../../src/AuthenticationParameters";
 import { ClientConfigurationError, ClientConfigurationErrorMessage } from "../../src/error/ClientConfigurationError";
 import { TEST_CONFIG, TEST_TOKEN_LIFETIMES } from "../TestConstants";
+import { ScopeSet } from "../../src/ScopeSet";
 import { StringDict } from "../../src/MsalTypes";
 import { TimeUtils } from "../../src/utils/TimeUtils";
 import sinon from "sinon";
-import { Constants } from "../../src/utils/Constants";
+import { Constants, InteractionType } from "../../src/utils/Constants";
 
+const clientId = TEST_CONFIG.MSAL_CLIENT_ID;
 
 describe("RequestUtils.ts class", () => {
+    describe("validateRequest", () => {
+        it("should throw emptyRequestError when the request passed in is null", () => {
+            let emptyRequestError: ClientConfigurationError;
+
+            try {
+                const userRequest: AuthenticationParameters = null;
+                const request: AuthenticationParameters = RequestUtils.validateRequest(userRequest, TEST_CONFIG.MSAL_CLIENT_ID, Constants.interactionTypeSilent);
+            } catch (e) {
+                emptyRequestError = e;
+            }
+
+            expect(emptyRequestError instanceof ClientConfigurationError).to.be.true;
+            expect(emptyRequestError.errorCode).to.equal(ClientConfigurationErrorMessage.emptyRequestError.code);
+            expect(emptyRequestError.name).to.equal("ClientConfigurationError");
+            expect(emptyRequestError.stack).to.include("RequestUtils.spec.ts");
+        });
+
+        it("should throw scopesRequiredError if scopes are empty or null", () => {
+            let scopesRequiredError: ClientConfigurationError;
+
+            try {
+                const userRequest: AuthenticationParameters = { };
```

Comment:
Instead of an empty request can we pass in a real request with just scopes missing? Should apply to any test specifically testing scope validation via `validateRequest` as this would break if the order of validation ever changes.

Reply:
I made the change for this specific test. The non-empty request I added only has a valid authority, which other request elements do you suggest I add? I'm not sure which of those would realistically be added in a token request other than scopes.

Reply:
I would suggest creating a complete and valid request object in a before or beforeEach block and then manipulate the particular field you are testing. e.g. Remove the scope field for this test.